File: src/main/java/org/example/packRLE.java
package org.example;

import javax.swing.text.html.parser.Parser;


import java.io.IOException;
import java.io.OutputStream;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;

public class packRLE  {






    public String packer(String s) {
        if (s.equals("")) {
            return "";
        }
        StringBuilder res = new StringBuilder();
        int count = 0;
        char prev = s.charAt(0);
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != prev) {
                if (count > 1) res.append(count);
                res.append(prev);
                count = 0;
                prev = s.charAt(i);
            }
            count++;
        }
        if (count > 1) res.append(count);
        res.append(prev);
        return res.toString();
    }


    public String unpacker(String str) {
        StringBuilder res = new StringBuilder();
        int num = 0;
        for (int i = 0; i < str.length(); i++) {
            if (i % 2 == 0) {
                num = str.charAt(i);
            } else {

                for (int k = 0; k < num - 1; k++) {
                    res.append(str.charAt(i));
                }
            }
        }
        return res.toString();
    }


    public static void pRLE(boolean pack, String inputFile, String outputFile) {



    }
}

File: src/main/java/org/example/packRLE.kt
package org.example

import java.io.File

class packRLE {




    fun packer(s: String): String {
        if (s == "") return ""
        val res = StringBuilder()
        var count = 0
        var prev = s[0]
        for (sym in s) {
            if (sym != prev) {
                if (count > 1) res.append(count)
                res.append(prev)
                count = 0
                prev = sym
            }
            count++
        }

        if (count > 1) res.append(count)
        res.append(prev)
        return res.toString()
    }


    fun unpacker(s: String): String {
        val res = StringBuilder()
        var i = 0
        while (i in s.indices) {
            val num = s.substring(i).takeWhile { it.isDigit() }
            val count = num.count()
            val char = s[i + count]

            repeat(num.toIntOrNull() ?: 1) { res.append(char) }
            i += 1 + count
        }
        return res.toString()
    }


    fun main(args: Array<String>) {
        Parser.main(args)
    }

    fun RLE(pack: Boolean, inputFile: String, outputFile: String?) {
        val inputStrings = File(inputFile).readLines()
        val outputStream = File(outputFile).bufferedWriter()
        outputStream.use {
            for (str in inputStrings) {
                val newStr = if (pack) packer(str) else unpacker(str)
                it.write(newStr)
                it.newLine()
            }
        }
        println("PackRLE: "+ if (pack) "pack" else "unpack")
    }

}
